test/org/sagebionetworks/bridge/validators/UploadSchemaValidatorTest.java

```diff
@@ -334,5 +326,32 @@ public void requiresFieldDefinitionHasType() throws Exception {
         schema = BridgeObjectMapper.get().readValue(json, UploadSchema.class);
         assertWillGenerateValidationError(schema, "fieldDefinitions0.name is required", "fieldDefinitions0.name");
     }
-    
+
+    @Test
+    public void duplicateFieldName() {
+        // set up schema to validate
+        DynamoUploadSchema schema = new DynamoUploadSchema();
+        schema.setName("Dupe Fields");
+        schema.setSchemaId("dupe-field-schema");
+        schema.setStudyId("test-study");
+        schema.setSchemaType(UploadSchemaType.IOS_SURVEY);
+
+        // test field def list
+        List<UploadFieldDefinition> fieldDefList = new ArrayList<>();
+        fieldDefList.add(new DynamoUploadFieldDefinition.Builder().withName("foo-field")
+                .withType(UploadFieldType.STRING).build());
+        fieldDefList.add(new DynamoUploadFieldDefinition.Builder().withName("foo-field")
+                .withType(UploadFieldType.INT).build());
+        schema.setFieldDefinitions(fieldDefList);
+
+        // validate
+        Exception thrownEx = null;
+        try {
+            Validate.entityThrowingException(UploadSchemaValidator.INSTANCE, schema);
+            fail("expected exception");
+        } catch (InvalidEntityException ex) {
+            thrownEx = ex;
+        }
+        assertTrue(thrownEx.getMessage().contains("foo-field"));
```

Comment:
Can you verify the message is correct and is understandable, e.g. UploadSchema is invalid: name 'foo-field' is used for more than one field definition?